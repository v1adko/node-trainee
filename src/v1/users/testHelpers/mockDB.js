import userDao from '../dao';
import userMocks from './userMocks';
import permissions from '../../../constants/permissions';

const { getUsersFields, user: userFields, admin: adminFields } = userMocks;

class MockDB {
  constructor(DAO) {
    this.DAO = DAO;
  }

  createDefaultUsers = async (userCount = 3) => {
    const users = [];
    getUsersFields(userCount).forEach(async (element) => {
      const { username, password, role } = element;
      const user = await this.createUser(username, password, role);
      users.push(user);
    });
  };

  createDefaultUser = async () => {
    const { username, password, role } = userFields;
    const user = await this.createUser(username, password, role);
    return user;
  };

  createDefaultAdmin = async () => {
    const { username, password, role } = adminFields;
    const user = await this.createUser(username, password, role);
    return user;
  };

  createUser = async (username, password, role = permissions.USER.value) => {
    const createdUser = this.DAO.create(username, password, role);
    return createdUser;
  };

  getAll = async () => {
    const users = await this.DAO.getAll();
    return users;
  };

  cleanDB = async () => {
    await this.DAO.Model.remove();
  };
}

export default new MockDB(userDao);
