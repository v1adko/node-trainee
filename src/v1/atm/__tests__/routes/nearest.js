import simulate from '../../../../utils/tests-utils/request-helper';
import UserFields from '../../../../utils/tests-utils/test-user-fields';
import User from '../../../users/user-model';
import mockDB from '../../../../utils/tests-utils/mock-db';
import atmUpdateService from '../../atm-update-service';

jest.setTimeout(50000);

const ROUTE = '/v1/atms/nearest';

const { getUserToken } = UserFields;
let userToken = null;

const setUp = async () => {
  await mockDB.cleanDB();
  userToken = getUserToken(User);
  await atmUpdateService.updateAtmsDataInDB();
};

const removeIdFromData = (data) => {
  const dataWithoutId = data;
  for (let i = 0; i < data.length; i += 1) {
    delete dataWithoutId[i]._id;
  }
  return dataWithoutId;
};

beforeAll(setUp);

afterAll(mockDB.closeConnection);

describe(`Test the ${ROUTE} path`, () => {
  it('should return 5 or less nearest atms (point is in Ukrain)', async () => {
    const route = `${ROUTE}?latitude=49.9935&longitude= 36.230383`;
    const result = await simulate.get(route, 200, userToken);
    const dataWithoutId = removeIdFromData(result);

    expect(dataWithoutId).toMatchSnapshot();
  });

  it('should return 5 or less nearest atms 1 (point is not in Ukrain)', async () => {
    const route = `${ROUTE}?latitude=-49.9935&longitude= 36.230383`;
    const result = await simulate.get(route, 200, userToken);
    const dataWithoutId = removeIdFromData(result);

    expect(dataWithoutId).toMatchSnapshot();
  });
});
