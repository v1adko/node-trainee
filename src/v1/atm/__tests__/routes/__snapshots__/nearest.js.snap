// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the /v1/atms/nearest path should return 5 or less nearest atms (point is in Ukrain) 1`] = `
Array [
  Object {
    "__v": 0,
    "cityEN": "Kharkiv",
    "cityRU": "Харьков",
    "cityUA": "Харкiв",
    "fullAddressEn": "Ukraine,area Kharkivska,city Kharkiv,square Konstytutsii,building 21/2",
    "fullAddressRu": "Украина,область Харьковская,город Харьков,площадь Конституции,дом 21/2",
    "fullAddressUa": "Украiна,область Харкiвська,мiсто Харкiв,площа Конституцii,будинок 21/2",
    "hash": "785cbc92f5dd3f7090b15d9dc08e7313625861fc",
    "latitude": 49.993631,
    "longitude": 36.230665,
    "placeRu": "Отделение Банка",
    "placeUa": "Вiддiлення банку",
    "tw": Object {
      "fri": "00:00 - 23:59",
      "hol": "00:00 - 23:59",
      "mon": "00:00 - 23:59",
      "sat": "00:00 - 23:59",
      "sun": "00:00 - 23:59",
      "thu": "00:00 - 23:59",
      "tue": "00:00 - 23:59",
      "wed": "00:00 - 23:59",
    },
    "type": "ATM",
  },
  Object {
    "__v": 0,
    "cityEN": "Kharkiv",
    "cityRU": "Харьков",
    "cityUA": "Харкiв",
    "fullAddressEn": "Ukraine,area Kharkivska,city Kharkiv,gatve Bursatskyi,building 4",
    "fullAddressRu": "Украина,область Харьковская,город Харьков,спуск Бурсацкий,дом 4",
    "fullAddressUa": "Украiна,область Харкiвська,мiсто Харкiв,узвiз Бурсацький,будинок 4",
    "hash": "394a4a28f290e5b7eb3bd606e262beaeaac3c16c",
    "latitude": 49.994026,
    "longitude": 36.229091,
    "placeRu": "Академия культуры",
    "placeUa": "",
    "tw": Object {
      "fri": "09:00 - 14:30",
      "hol": "00:00 - 00:00",
      "mon": "09:00 - 15:00",
      "sat": "00:00 - 00:00",
      "sun": "00:00 - 00:00",
      "thu": "09:00 - 15:00",
      "tue": "09:00 - 15:00",
      "wed": "09:00 - 15:00",
    },
    "type": "ATM",
  },
  Object {
    "__v": 0,
    "cityEN": "Kharkiv",
    "cityRU": "Харьков",
    "cityUA": "Харкiв",
    "fullAddressEn": "Ukraine,area Kharkivska,city Kharkiv,square Konstytutsii,building 11/13",
    "fullAddressRu": "Украина,область Харьковская,город Харьков,площадь Конституции,дом 11/13",
    "fullAddressUa": "Украiна,область Харкiвська,мiсто Харкiв,площа Конституцii,будинок 11/13",
    "hash": "ee75a32f279c4c788ab5366720028981eb3a1353",
    "latitude": 49.991486,
    "longitude": 36.231739,
    "placeRu": "Университет искусств, кафе",
    "placeUa": "Iнстiтут мистецтв",
    "tw": Object {
      "fri": "09:00 - 19:00",
      "hol": "00:00 - 00:00",
      "mon": "09:00 - 19:00",
      "sat": "09:00 - 18:00",
      "sun": "09:00 - 18:00",
      "thu": "09:00 - 19:00",
      "tue": "09:00 - 19:00",
      "wed": "09:00 - 19:00",
    },
    "type": "ATM",
  },
  Object {
    "__v": 0,
    "cityEN": "Kharkiv",
    "cityRU": "Харьков",
    "cityUA": "Харкiв",
    "fullAddressEn": "Ukraine,area Kharkivska,city Kharkiv,street Donets-Zakharzhevskoho,building 6/8",
    "fullAddressRu": "Украина,область Харьковская,город Харьков,улица Донец-Захаржевского,дом 6/8",
    "fullAddressUa": "Украiна,область Харкiвська,мiсто Харкiв,вулиця Донець-Захаржевського,будинок 6/8",
    "hash": "e54f2e3a11460292972a64bc0e97389f4929901a",
    "latitude": 49.993426,
    "longitude": 36.23306,
    "placeRu": "Отделение Банка",
    "placeUa": "",
    "tw": Object {
      "fri": "00:00 - 23:59",
      "hol": "00:00 - 23:59",
      "mon": "00:00 - 23:59",
      "sat": "00:00 - 23:59",
      "sun": "00:00 - 23:59",
      "thu": "00:00 - 23:59",
      "tue": "00:00 - 23:59",
      "wed": "00:00 - 23:59",
    },
    "type": "ATM",
  },
]
`;

exports[`Test the /v1/atms/nearest path should return 5 or less nearest atms 1 (point is not in Ukrain) 1`] = `
Array [
  Object {
    "__v": 0,
    "cityEN": "Kharkiv",
    "cityRU": "Харьков",
    "cityUA": "Харкiв",
    "fullAddressEn": "Ukraine,area Kharkivska,city Kharkiv,prospect Yuriia Haharina,building 352",
    "fullAddressRu": "Украина,область Харьковская,город Харьков,проспект Юрия Гагарина,дом 352",
    "fullAddressUa": "Украiна,область Харкiвська,мiсто Харкiв,проспект Юрiя Гагарiна,будинок 352",
    "hash": "43b4329acb97a78dfa8ce7104ba82efa058e2ead",
    "latitude": 49.909588,
    "longitude": 36.275419,
    "placeRu": "Гипермаркет \\"Эпицентр\\"",
    "placeUa": "Гiпермаркет \\"Епiцентр\\"",
    "tw": Object {
      "fri": "08:00 - 21:00",
      "hol": "08:00 - 21:00",
      "mon": "08:00 - 21:00",
      "sat": "08:00 - 21:00",
      "sun": "08:00 - 21:00",
      "thu": "08:00 - 21:00",
      "tue": "08:00 - 21:00",
      "wed": "08:00 - 21:00",
    },
    "type": "ATM",
  },
  Object {
    "__v": 0,
    "cityEN": "Kharkiv",
    "cityRU": "Харьков",
    "cityUA": "Харкiв",
    "fullAddressEn": "Ukraine,area Kharkivska,city Kharkiv,building 25",
    "fullAddressRu": "Украина,область Харьковская,город Харьков,шоссе Мерефянское,дом 25",
    "fullAddressUa": "Украiна,область Харкiвська,мiсто Харкiв,шосе Мереф''янське,будинок 25",
    "hash": "08e0de1d0969f271e1ea829c413f99998b2777ea",
    "latitude": 49.911827,
    "longitude": 36.22425,
    "placeRu": "Магазин Вилена",
    "placeUa": "Магазин \\"Вiленд\\"",
    "tw": Object {
      "fri": "07:00 - 22:00",
      "hol": "07:00 - 22:00",
      "mon": "07:00 - 22:00",
      "sat": "07:00 - 22:00",
      "sun": "07:00 - 22:00",
      "thu": "07:00 - 22:00",
      "tue": "07:00 - 22:00",
      "wed": "07:00 - 22:00",
    },
    "type": "ATM",
  },
  Object {
    "__v": 0,
    "cityEN": "Kharkiv",
    "cityRU": "Харьков",
    "cityUA": "Харкiв",
    "fullAddressEn": "Ukraine,area Kharkivska,city Kharkiv,street Bezliudivska,building 2-b",
    "fullAddressRu": "Украина,область Харьковская,город Харьков,улица Безлюдовская,дом 2-б",
    "fullAddressUa": "Украiна,область Харкiвська,мiсто Харкiв,вулиця Безлюдiвська,будинок 2-б",
    "hash": "990f06d73d9ed85a4ac34197599ce674fe6ab63d",
    "latitude": 49.913036,
    "longitude": 36.277402,
    "placeRu": "Проходная фабрики \\"Nestle\\"",
    "placeUa": "Прохiдна фабрики \\"Nestle\\"",
    "tw": Object {
      "fri": "08:00 - 19:00",
      "hol": "00:00 - 00:00",
      "mon": "08:00 - 19:00",
      "sat": "08:00 - 17:00",
      "sun": "00:00 - 00:00",
      "thu": "08:00 - 19:00",
      "tue": "08:00 - 19:00",
      "wed": "08:00 - 19:00",
    },
    "type": "ATM",
  },
  Object {
    "__v": 0,
    "cityEN": "Kharkiv",
    "cityRU": "Харьков",
    "cityUA": "Харкiв",
    "fullAddressEn": "Ukraine,area Kharkivska,city Kharkiv,street Hrytsevtsia,building 35-a",
    "fullAddressRu": "Украина,область Харьковская,город Харьков,улица Грицевца,дом 35-а",
    "fullAddressUa": "Украiна,область Харкiвська,мiсто Харкiв,вулиця Грицевця,будинок 35-а",
    "hash": "95ba8766c993ba2dda128225bf38d42a4aa5ffce",
    "latitude": 49.915544,
    "longitude": 36.408101,
    "placeRu": "Отделение Банка",
    "placeUa": "Вiддiлення банку",
    "tw": Object {
      "fri": "00:00 - 23:59",
      "hol": "00:00 - 23:59",
      "mon": "00:00 - 23:59",
      "sat": "00:00 - 23:59",
      "sun": "00:00 - 23:59",
      "thu": "00:00 - 23:59",
      "tue": "00:00 - 23:59",
      "wed": "00:00 - 23:59",
    },
    "type": "ATM",
  },
]
`;
